src/cache.js

```diff
@@ -16,28 +16,73 @@ const stringToId = str => {
 
   return str
 }
+const fieldToDocField = key => (key === 'id' ? '_id' : key)
 
 // https://github.com/graphql/dataloader#batch-function
-const orderDocs = ids => docs => {
-  const idMap = {}
-  docs.forEach(doc => {
-    idMap[idToString(doc._id)] = doc
-  })
-  return ids.map(id => idMap[idToString(id)])
-}
+// "The Array of values must be the same length as the Array of keys."
+// "Each index in the Array of values must correspond to the same index in the Array of keys."
+const orderDocs = fieldsArray => docs =>
+  fieldsArray.map(fields =>
+    docs.filter(doc => {
+      for (let fieldName of Object.keys(fields)) {
+        const fieldValue = fields[fieldName]
+        if (typeof fieldValue === 'undefined') continue
+        const filterValuesArr = Array.isArray(fieldValue)
+          ? fieldValue.map(val => idToString(val))
+          : [idToString(fieldValue)]
+        const docValue = doc[fieldToDocField(fieldName)]
+        const docValuesArr = Array.isArray(docValue)
+          ? docValue.map(val => idToString(val))
+          : [idToString(docValue)]
+        let isMatch = false
+        for (const filterVal of filterValuesArr) {
+          if (docValuesArr.includes(filterVal)) {
+            isMatch = true
+          }
+        }
+        if (!isMatch) return false
+      }
+      return true
+    })
+  )
 
 export const createCachingMethods = ({ collection, model, cache }) => {
-  const loader = new DataLoader(ids => {
-    const filter = {
-      _id: {
-        $in: ids.map(stringToId)
+  const loader = new DataLoader(JSONArray => {
+    const fieldsArray = JSONArray.map(JSON.parse)
```

Comment:
nit pick: :man_shrugging: 

```suggestion
  const loader = new DataLoader(jsonArray => {
    const fieldsArray = jsonArray.map(JSON.parse)
```

`JSONArray` looks like a class name.

Reply:
Good call. Fixed.